src/smc-webapp/jupyter/top-buttonbar.tsx

```diff
@@ -187,39 +197,53 @@ export class TopButtonbar0 extends Component<TopButtonbarProps> {
     ]);
   }
 
-  cell_select_type(event: any): void {
-    this.props.frame_actions.set_selected_cell_type(event.target.value);
+  cell_select_type(key: any): void {
+    this.props.frame_actions.set_selected_cell_type(key);
     this.focus();
   }
 
-  render_select_cell_type() {
-    let cell_type: any;
-    if (this.props.sel_ids.size > 1) {
-      cell_type = "multi";
-    } else {
-      cell_type = this.props.cells.getIn(
-        [this.props.cur_id, "cell_type"],
-        "code"
-      );
+  private static cell_type_title(key): string {
+    switch (key) {
+      case "multi":
+        return "-";
+      default:
+        return capitalize(key);
     }
+  }
+
+  render_select_cell_type() {
+    const cell_type = (() => {
+      if (this.props.sel_ids.size > 1) {
+        return "multi";
+      } else {
+        return this.props.cells.getIn([this.props.cur_id, "cell_type"], "code");
+      }
+    })();
```

Comment:
Did you use an IIFE just for the automatic type assignment to `cell_type`? This seems cool I think?

Reply:
well, yes, I guess so. `cell_type : any` looked to me as if it needs some improval.

however, speaking of this in particular, I tested it and think there is a bug → https://github.com/sagemathinc/cocalc/issues/4244

I haven't checked, but I can't believe this is caused by this PR.

Reply:
it's also in prod

Reply:
This is prob just decaffeinate ugliness left over from Travis's conversion from Summer 2018...

Reply:
>it's also in prod 

Oh, you meant the bug is in prod.

@williamstein The diff suggests this is code @haraldschilly wrote, not decaffeinate.

Reply:
const instead let

Reply:
everybody would be ok with using ? : instead for this particular goal.